marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -437,6 +439,18 @@ function addPrimaryKeyToResult(result, entityInstance, entityDef) {
   }
 }
 
+function removeEntityNameFromCollection(searchResponse, entityName) {
+  if (searchResponse.hasOwnProperty('facets')) {
```

Comment:
This is fine for now, but really, a custom constraint would be better here. That's because it would avoid having ML calculate the number of documents in the entity-name collection, which would speed up search. 

But really, the right long term solution is to not have a collection facet. That's on my kill list for 5.4.0. 

Pinging @fsnow  about this too, just for awareness.